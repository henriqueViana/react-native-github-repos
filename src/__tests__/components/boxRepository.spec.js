import React from 'react'
import { mount } from 'enzyme'

import BoxRepository from '../../components/boxRepository'

let repository
let wrapper

describe('-----BoxRepository-----', () => {

  beforeEach(() => {
    repository = {
      id: 1,
      name: 'nameRepo',
      description: 'descriptionRepo'
    }

    wrapper = mount(<BoxRepository repository={repository}/>)
  })

  it('should render correcty BoxRepository', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a repository property', () => {
    expect(wrapper.props().repository).toMatchObject(repository)
  })
})