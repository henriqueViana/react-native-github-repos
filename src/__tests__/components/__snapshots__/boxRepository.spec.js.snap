// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-----BoxRepository----- should render correcty BoxRepository 1`] = `
<BoxRepository
  repository={
    Object {
      "description": "descriptionRepo",
      "id": 1,
      "name": "nameRepo",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "borderColor": "#CCC",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#FFF",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0,
            "flexGrow": 5,
            "width": 100,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "flexGrow": 5,
              "width": 100,
            }
          }
        >
          <Text>
            <Text>
              nameRepo
            </Text>
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#ccc",
                "fontSize": 13,
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#ccc",
                  "fontSize": 13,
                  "marginTop": 5,
                }
              }
            >
              descriptionRepo
            </Text>
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 0,
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0,
              "flexGrow": 1,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#000"
            name="chevron-right"
            size={25}
          >
            <Text>
              <Text />
            </Text>
          </Icon>
        </View>
      </View>
    </View>
  </View>
</BoxRepository>
`;
